From 9fd5160f293001a6e318c161dcc980992735a34f Mon Sep 17 00:00:00 2001
From: Flant JSC <support@flant.com>
Date: Mon, 22 Sep 2026 21:00:00 +0300
Subject: [PATCH] Expose TLS iPXE endpoint when Ironic TLS is enabled

The ironic image enables its iPXE TLS vhost when the serving certificate is
available at /certs/ipxe. Mount the existing Ironic TLS Secret there and expose
the corresponding host-network port.

---
 pkg/ironic/containers.go | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/pkg/ironic/containers.go b/pkg/ironic/containers.go
index 0f4d1b2..489cd72 100644
--- a/pkg/ironic/containers.go
+++ b/pkg/ironic/containers.go
@@ -34,6 +34,7 @@ const (
 	authDir   = "/auth"
 	certsDir  = "/certs"
 	sharedDir = "/shared"
+	ipxeTLSPort = 8084
 	metricsPort            = 9608
 	defaultMetricsBindAddr = "0.0.0.0"
@@ -534,6 +535,12 @@ func buildIronicVolumesAndMounts(resources Resources) (volumes []corev1.Volume,
 				ReadOnly:  true,
 			},
 		)
+		mounts = append(mounts, corev1.VolumeMount{
+			Name:      "cert-ironic",
+			MountPath: certsDir + "/ipxe",
+			ReadOnly:  true,
+		})
 	}
 	if maybeVolume := volumeForSecretOrConfigMap(bmcCAVolumeName, resources.BMCCASecret, resources.BMCCAConfigMap); maybeVolume != nil {
@@ -591,6 +598,12 @@ func buildIronicHttpdPorts(ironic *metal3api.Ironic) (ironicPorts []corev1.Cont
 				HostPort:      ironic.Spec.Networking.ImageServerTLSPort,
 			})
 		}
+		httpdPorts = append(httpdPorts, corev1.ContainerPort{
+			Name:          "ipxe-tls",
+			Protocol:      corev1.ProtocolTCP,
+			ContainerPort: ipxeTLSPort,
+			HostPort:      ipxeTLSPort,
+		})
 	}
 
 	return ironicPorts, httpdPorts
